calculator.go

```diff
@@ -1,13 +1,125 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+)
+
+// Add takes a list of numbers and returns the result of adding them together.
+func Add(nums ...float64) float64 {
+	switch len(nums) {
+	case 0:
+		return 0
+	case 1:
+		return nums[0]
+	default:
+		var sum float64
+		for _, n := range nums {
+			sum += n
+		}
+		return sum
+	}
+}
+
+// Subtract takes a list of numbers and returns the result of subtracting them from each other.
+func Subtract(nums ...float64) float64 {
+	switch len(nums) {
+	case 0:
+		return 0
+	case 1:
+		return nums[0]
+	default:
+		var sum float64 = nums[0]
+		for _, n := range nums[1:] {
+			sum -= n
+		}
+		return sum
+	}
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
-// from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+// Multiply takes a list of numbers and multiplies them.
+func Multiply(nums ...float64) float64 {
+	switch len(nums) {
+	case 0:
+		return 0
+	case 1:
+		return nums[0]
+	default:
+		var product float64 = nums[0]
+		for _, n := range nums[1:] {
+			product *= n
+		}
+		return product
+	}
+}
+
+// Divide takes a list of numbers and divides them.
+// Error is returned when dividing by 0.
+func Divide(nums ...float64) (float64, error) {
+	switch len(nums) {
+	case 0:
+		return 0, nil
+	case 1:
+		return nums[0], nil
+	default:
+		var product float64 = nums[0]
```

Comment:
`product` doesn't seem like quite the right name, does it? How about something more general like `result`?